### PR TITLE
Update check line alignment to check by tag

### PR DESCRIPTION
**Pull request by renatho**
This PR is temporarily based on https://github.com/gajus/eslint-plugin-jsdoc/pull/685 until it gets merged.

This PR updates the check line alignment to check by tag (when using the `always` option) instead of the whole block, as suggested here: https://github.com/gajus/eslint-plugin-jsdoc/issues/680#issuecomment-770314024.

It also adds support to the `tags` option.

@brettz9, I wasn't able to finish that before my vacation. So I'm opening this PR with the current changes I did so far. I think it's very close, but it still needs some details.

My approach is cloning the `jsdoc`, parse that, and then get the aligned positions fixing the original one only in the _applicable tags_. The issue is that the cloned comment will align every line, so it impacts the position of the other alignments. So in a case with `typedef`, for example, if we wanna align only the `property` tags, we can have something like:

```
        /**
         * My object.
         *
         * @typedef {Object} MyObjectWithABigNameBlaBlaBla
         *
         * @property {string} lorem Description.
         * @property {int} sit Description multi words.
         */
```

**Output:**

```
        /**
         * My object.
         *
         * @typedef  {Object} MyObjectWithABigNameBlaBlaBla
         *
         * @property {string}  lorem                        Description.
         * @property {int}       sit                        Description multi words.
         */
```

If someone wants to get this PR to finish the work, feel free to. Otherwise, it'll take a while until I'm able to work on that again. 😉

### CODE REVIEW

File: src/iterateJsdoc.js [hunk 1]
```diff
@@ -157,6 +159,8 @@ const getBasicUtils = (context, {tagNamePreference, mode}) => {
     return ret;
   };
 
+  utils.commentParser = parseComment;
```
Comment (renatho) [inline src/iterateJsdoc.js:162]: Not sure if we really want to expose that, but I did it to use the same method to parse a new comment to get the correct alignments for comparison.

Comment (brettz9) [reply src/iterateJsdoc.js:162]: Works for me!

**Comment by renatho** [comment]
An idea to fix the pending issues of this PR is removing the tags we don't want align before parsing the clone.

**Comment by renatho** [comment]
Since the original branch was already merged, a new PR was opened pointing to the original repo: https://github.com/gajus/eslint-plugin-jsdoc/pull/689

**Pull request closed without merging**